File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.ts
const postorderTraversal = (root: TreeNode | null): number[] => {
  const values: number[] = [];
  const traverse = (node: TreeNode): void => {
    if (!node) return;
    traverse(node.left);
    traverse(node.right);
    values.push(node.val);
  };
  traverse(root);
  return values;
};

// const postorderTraversal = (root: TreeNode | null): number[] => {
//   const values: number[] = [];
//   const traverse = (node: TreeNode): void => {
//     if (node.left) traverse(node.left);
//     if (node.right) traverse(node.right);
//     values.push(node.val);
//   };
//   if (root !== null) traverse(root);
//   return values;
// };

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
  def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
    ans = []
    def traverse(node):
      if node.left:
        traverse(node.left)
      if node.right:
        traverse(node.right)
      ans.append(node.val)
    if root:
      traverse(root)
    return ans
        